subprojects/studio-plugin/src/main/java/org/gradle/profiler/studio/plugin/system/AndroidStudioSystemHelper.java

```diff
@@ -88,11 +89,13 @@ public static void waitOnPreviousGradleSyncFinish(Project project) {
      * It seems there is no better way to do it atm.
      */
     public static void waitOnBackgroundProcessesFinish(Project project) {
-        IdeFrame frame = WindowManagerEx.getInstanceEx().findFrameFor(project);
-        StatusBarEx statusBar = frame == null ? null : (StatusBarEx) frame.getStatusBar();
-        if (statusBar != null) {
-            statusBar.getBackgroundProcesses().forEach(it -> waitOnProgressIndicator(it.getSecond()));
-        }
+        DumbService.getInstance(project).runReadActionInSmartMode(() -> {
+            IdeFrame frame = WindowManagerEx.getInstanceEx().findFrameFor(project);
+            StatusBarEx statusBar = frame == null ? null : (StatusBarEx) frame.getStatusBar();
+            if (statusBar != null) {
+                statusBar.getBackgroundProcesses().forEach(it -> waitOnProgressIndicator(it.getSecond()));
+            }
+        });
```

Comment:
❓ Are we losing compatibility with some older IntelliJ/Android Studio versions by switching to this API? If so, can we document how the compatibility changes?

Reply:
Yes, we are compatible with older versions. We actually still compile our plugin with IntelliJ 2021.1.1 API (Android Studio Chipmunk), so with the first version we added profiler support.